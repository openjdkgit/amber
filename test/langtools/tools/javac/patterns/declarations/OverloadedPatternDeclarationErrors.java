/*
 * @test /nodynamiccopyright/
 * @summary Verify error related to annotations and patterns
 * @enablePreview
 * @compile/fail/ref=OverloadedPatternDeclarationErrors.out -XDrawDiagnostics -XDdev OverloadedPatternDeclarationErrors.java
 */
public class OverloadedPatternDeclarationErrors {
    private static int test(D o) {
        if (o instanceof D(String data, Integer out)) { // no compatible matcher found
            return out;
        }
        return -1;
    }

    private static int test2(D2 o) {
        if (o instanceof D2(String data, Integer out)) { // ambiguous
            return out;
        }
        return -1;
    }

    public static class D {
        public pattern D(Object v1, Float out) {
            match D(10.0f, 10.0f);
        }

        public pattern D(Float out, Integer v1) {
            match D(10.0f, 2);
        }
    }

    public static class D2 {
        public pattern D2(Object v, Integer out) {
            match D2("", 1);
        }

        public pattern D2(CharSequence v, Integer out) {
            match D2("", 2);
        }
    }
}
