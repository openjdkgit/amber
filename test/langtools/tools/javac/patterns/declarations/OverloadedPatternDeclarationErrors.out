OverloadedPatternDeclarationErrors.java:9:26: compiler.err.no.compatible.matcher.found
OverloadedPatternDeclarationErrors.java:16:26: compiler.err.matcher.overloading.ambiguity
- compiler.note.preview.filename: OverloadedPatternDeclarationErrors.java, DEFAULT
- compiler.note.preview.recompile
2 errors